### PR TITLE
Add initial gemspec

### PR DESCRIPTION
**Pull request by colinbendell**
This addresses issues #88 by adding a simple gemspec for other projects to consume

### CODE REVIEW

File: protoboeuf.gemspec [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  spec.summary = "A pure-Ruby protobuf encoder / decoder supporting `proto3`"
+  spec.description = spec.summary
+  spec.homepage = "https://github.com/Shopify/protoboeuf"
+
+  spec.metadata["homepage_uri"] = spec.homepage
+  spec.metadata["source_code_uri"] = spec.homepage
+  spec.metadata["changelog_uri"] = "#{spec.homepage}/releases"
+
+  spec.bindir = "bin"
```
[the remark this answers is not in the document]
Comment (maximecb) [reply protoboeuf.gemspec:20]: @tenderworks should we move `bin/protoboeuf` to `exe/protoboeuf`? 🤔

File: protoboeuf.gemspec [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  spec.executables = ["protoboeuf"]
+  spec.require_paths = ["lib"]
+  spec.files = Dir.glob(%w[lib/**/*.rb]) + %w[README.md Gemfile Rakefile]
+
+  spec.metadata["allowed_push_host"] = "https://rubygems.org"
+
+  spec.add_dependency("syntax_tree")
+
+  spec.required_ruby_version = ">= 3.3"
```
[the remark this answers is not in the document]
Comment (maximecb) [reply protoboeuf.gemspec:29]: We actually test with 3.2 in the CI, so this could be changed to 3.2.

[the remark this answers is not in the document]
Comment (casperisfine) [reply protoboeuf.gemspec:29]: Actually, it's tested on CI with Ruby 3.2, and I tried it locally with 3.1, works fine. I doubt there's really anything in there requiring to force a recent Ruby version.

Maybe asking for 3.0 make sense, but not much more.

File: bin/protoboeuf [hunk 3]
```diff
@@ -1,13 +1,14 @@
-#!env ruby
+#!/usr/bin/env ruby
 # frozen_string_literal: true
 
-require_relative "../lib/protoboeuf/codegen"
-require_relative "../lib/protoboeuf/parser"
+lib = File.expand_path("../../lib", __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
```
[the remark this answers is not in the document]
Comment (colinbendell) [reply bin/protoboeuf:5]: yea, it's just awkward to run the script locally and not in the gem. Is there a better pattern to support both gem and local invocation?

[the remark this answers is not in the document]
Comment (maximecb) [reply bin/protoboeuf:5]: Can you do a begin rescue on the load?

**Comment by maximecb** [comment]
Thanks everyone :D

**Pull request merged**